### PR TITLE
Inline Layer 3 is a remote network of an Inline Layer 2 - New filter based on the network (cidr representation)

### PR DESCRIPTION
**Pull request by fdurand**
For review and merge.

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Model/Interface.pm [hunk 1]
```diff
@@ -715,9 +713,38 @@ sub ACCEPT_CONTEXT {
     return $object;
 }
 
+=head2 getEnforcement
+
+=cut
+
+sub getEnforcement {
```
Comment (cgx) [inline html/pfappserver/lib/pfappserver/Model/Interface.pm:721]: Can't we reuse the `getType` function? Or call `getEnforcement` from `getType`? Please avoid duplicated code.

Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Model/Interface.pm:721]: Type is not the same thing than Enforcement and what i need is the enforcement value in pf.conf, not from networks.conf

Comment (cgx) [inline html/pfappserver/lib/pfappserver/Model/Interface.pm:721]: I misread the code. Keep it like this. Thanks.

File: NEWS.asciidoc [hunk 2]
```diff
@@ -18,11 +18,12 @@ New Features
 ++++++++++++
 
 * Added the possibility to search by computer name on the nodes page
+* New portal profile filter based on the ip address of the device
```
Comment (extrafu) [inline NEWS.asciidoc]: New portal profile filter based on ranges of device's IP addresses

Would that be a better description?

File: html/pfappserver/lib/pfappserver/Form/Field/ProfileFilter.pm [hunk 3]
```diff
@@ -49,6 +49,7 @@ has_field 'type' =>
                { value => 'vlan', label => 'VLAN' },
                { value => 'switch', label => 'SWITCH' },
                { value => 'uri', label => 'URI' },
+               { value => 'ip', label => 'NETWORK' },
```
Comment (jrouzierinverse) [inline html/pfappserver/lib/pfappserver/Form/Field/ProfileFilter.pm]: 'ip' should be change to either 'network' or 'cidr'

File: lib/pf/Portal/ProfileFactory.pm [hunk 4]
```diff
@@ -33,7 +35,7 @@ Create a new pf::Portal::Profile instance based on parameters given.
 
 =cut
 
-our @MATCHES_TYPE = qw(uri ssid vlan switch);
+our @MATCHES_TYPE = qw(uri ssid vlan switch ip);
```
Comment (jrouzierinverse) [inline lib/pf/Portal/ProfileFactory.pm]: 'ip' should be change to either 'network' or 'cidr'

File: lib/pf/Portal/ProfileFactory.pm [hunk 5]
```diff
@@ -52,6 +54,16 @@ sub instantiate {
     my $node_info = node_view($mac) || {};
     $node_info = { %$options, %$node_info } ;
 
+    foreach my $filter (keys %Profile_Filters) {
+        if ($filter =~ /ip:(.*)/) {
```
Comment (jrouzierinverse) [inline lib/pf/Portal/ProfileFactory.pm]: 'ip' should be change to either 'network' or 'cidr'

File: NEWS.asciidoc (hunk 2 above)
Comment (fdurand) [inline NEWS.asciidoc]: Or maybe something like : 
New portal profile based on Network range (ex: 192.168.0.0/24)

File: lib/pf/Portal/ProfileFactory.pm (hunk 4 above)
Comment (fdurand) [inline lib/pf/Portal/ProfileFactory.pm]: I vote for network, even if it's the ip address of the device we test, we change it on the fly by the network address.

**Pull request merged**